lib/puppet/util/windows/sid.rb

```diff
@@ -68,6 +68,11 @@ def name_to_principal(name)
       # Apparently, we accept a symbol..
       name = name.to_s.strip if name
```

Comment:
Having this code here, if `name` is already a `Principal` we will convert it to string and back to `Principal`.
Ans this conversion is not idempotent is case of an `unresolvable SID`